crates/db_schema/src/impls/local_user.rs

```diff
@@ -115,11 +115,11 @@ impl LocalUser {
     let conn = &mut get_conn(pool).await?;
 
     // Make sure:
-    // - The deny reason exists
+    // - An admin has interacted with the application
     // - The app is older than a week
     // - The accepted_application is false
     let old_denied_registrations = registration_application::table
-      .filter(registration_application::deny_reason.is_not_null())
+      .filter(registration_application::admin_id.is_not_null())
```

Comment:
This would also delete approved applications, which isn't a good thing. 

We don't have a great way to view them currently, but I could see it being useful later to allow viewing of a local user's registration application before taking mod action on them (maybe allow admins to view it from their profile page, I'll open up an issue for that).

I agree that the best way to handle it logically would be to make sure deny_reason is required in the API when `data.approved = false`

Reply:
How would this delete approved applications?
This doesn't delete applications, it deletes persons that have `accepted_application` set to `false`.
Deletion of applications is cascaded from foreign keys from the deletion of the person.

The SQL query is not super great to read due to diesel [not allowing joins for deletes](https://github.com/diesel-rs/diesel/issues/1478), but `old_denied_registrations` and `local_users` are just subqueries that are fed into `DELETE FROM person`.

If the transaction part that @dullbananas pointed out was fixed there wouldn't be a race condition (or potential failure of the second query), but we already do have reliable logic to identify whether an application is accepted or rejected.

An application initially has no `admin_id` associated with it.
This is how you can identify that nobody has interacted with the application yet.

If email verification is enabled on the instance we hide applications from admins until the persons email has been verified.

Once an admin interacts with the application, `admin_id` gets set and the application can only have two states: accepted or denied.
It is not possible to revert an application to the untouched state after approving or denying it before. I also can't think of a good reason for when you'd want to do this.
This state is now indicated by `person.accepted_application` being `true` or `false`.

So effectively we have total of 4 possible states of an application:

1. Application is created but email is not verified when email verification is required.
2. Application is created and email is verified or email verification is not required.
3. Application has been accepted
4. Application has been denied

All of these can already be reliably identified.

Reply:
> This doesn't delete applications, it deletes persons that have accepted_application set to false.
Deletion of applications is cascaded from foreign keys from the deletion of the person.

Ah I see you're correct there. In that case this is probably fine.

Reply:
I discovered another unexpected case while implementing the tests in #4923, which I raised #4969 for:

Applications can be pending (no admin id set) even when the user has `accepted_application` set to `true`.

This however does not have any impact on the deletion logic, as this query verifies that an `admin_id` is set on the application and the query following this one validates that the user must have `accepted_application` set to `false`, so there are already two conditions that these cases would fail.